commit 7f8f488a5a526dee7f8e7fe57a99c49fc8eb79d9
Author: Dmitry Grinberg <dmitrygr@gmail.com>
Date:   Mon Dec 26 23:50:03 2022 -0600

    do not misuse FtrPtrFree to free dynamic heap chunks, it is wrong

diff --git a/Src/Palmkedex.c b/Src/Palmkedex.c
index c70b85a..2822ca1 100644
--- a/Src/Palmkedex.c
+++ b/Src/Palmkedex.c
@@ -231,7 +231,12 @@ static Err AppStart(void)
 
 static void UnloadSpecies()
 {
-	FtrPtrFree(appFileCreator, ftrPkmnNamesNum);
+	void *ptr;
+	
+	if (errNone == FtrGet(appFileCreator, ftrPkmnNamesNum, (UInt32*)&ptr))
+		MemPtrFree(ptr);
+	
+	FtrUnregister(appFileCreator, ftrPkmnNamesNum);
 }
 
 static void FreeSharedVariables()
@@ -258,8 +263,8 @@ static void FreeSharedVariables()
 	{
 		MemPtrFree(sharedVars->pkmnFormTitle);
 	}
-	
-	FtrPtrFree(appFileCreator, ftrShrdVarsNum);
+	MemPtrFree(sharedVars);
+	FtrUnregister(appFileCreator, ftrShrdVarsNum);
 }
 
 /*
